docs/content/config/advanced/kubernetes.md

```diff
@@ -4,518 +4,626 @@ title: 'Advanced | Kubernetes'
+## Exposing your Mail Server to the Outside World
 
-!!! attention "Sensitive Data"
+The more difficult part with Kubernetes is to expose a deployed DMS instance to the outside world. Kubernetes provides multiple ways for doing that; each has its upsides and downsides. The major problem with exposing DMS to the outside world in Kubernetes is to [preserve the real client IP][Kubernetes-service-source-ip]. The real client IP is required by DMS for performing IP-based DNS and spam checks.
 
-    For storing OpenDKIM keys, TLS certificates or any sort of sensitive data, you should be using `Secret`s. You can mount secrets like `ConfigMap`s and use them the same way.
+=== "Load-Balancer + Public IP"
 
-The [TLS docs page][docs-tls] provides guidance when it comes to certificates and transport layer security. Always provide sensitive information vai `Secrets`.
+    This approach only works when you have a **dedicated** IP address that you can give to the responsible `Service`, e.g., with a load balancer like [MetalLB][metallb-web]. Such an IP has to be public and therefore routable. The IP is required to be dedicated to allow your mail server to have matching `A` and `PTR` records (that other mail server can checken when you send them e-mails).
 
-## Exposing your Mail Server to the Outside World
+    The upside is that the manifests files and the configuration do not become more complex; the downside is that you require a dedicated IPv4 address and you are stuck to the node that has this IP address bound.
 
-The more difficult part with Kubernetes is to expose a deployed DMS to the outside world. Kubernetes provides multiple ways for doing that; each has downsides and complexity. The major problem with exposing DMS to outside world in Kubernetes is to [preserve the real client IP][Kubernetes-service-source-ip]. The real client IP is required by DMS for performing IP-based SPF checks and spam checks. If you do not require SPF checks for incoming mails, you may disable them in your [Postfix configuration][docs-postfix] by dropping the line that states: `check_policy_service unix:private/policyd-spf`.
+=== "External-IP Service"
 
-The easiest approach was covered above, using `#!yaml externalTrafficPolicy: Local`, which disables the service proxy, but makes the service local as well (which does not scale). This approach only works when you are given the correct (that is, a public and routable) IP address by a load balancer (like MetalLB). In this sense, the approach above is similar to the next example below. We want to provide you with a few alternatives too. **But** we also want to communicate the idea of another simple method: you could use a load-balancer without an external IP and DNAT the network traffic to the mail server. After all, this does not interfere with SPF checks because it keeps the origin IP address. If no dedicated external IP address is available, you could try the latter approach, if one is available, use the former.
```

Comment:
`externalTrafficPolicy` was actually outdated because it was removed entirely with an earlier PR.

> But now some of that context is missing with changes proposed by this PR?

I removed what I thought to be superfluous; the information in this section is accurate now, but shorter. I am not worried about this.

> Presumably this is part of the "Load-Balancer + Public IP" tab approach, as the paragraph then continues to refer to the same approach but stating the public routable IP requirement for a load balancer and using MetalLB as an example. It refers to "approach above" and the "next example" (now "External-IP Service" tab) being similar, although I'm not getting much value as the reader from that.

I have rephrased one paragraph a bit now; but where is the "next example" line? I could not find it...

Reply:
> where is the "next example" line? I could not find it...

![image](https://github.com/docker-mailserver/docker-mailserver/assets/5098581/b569c280-64ab-4fe7-9609-4defb39f736d)

Reply:
Ah, it was in the already removed diff; thanks for the clarification!